src/Templating/Helper/BlockHelper.php

```diff
@@ -115,142 +87,16 @@ class BlockHelper
      */
     private $stopwatch;
 
-    /**
-     * NEXT_MAJOR: remove the deprecated signature and cleanup the constructor.
-     *
-     * @param array{by_class: array<class-string, string>, by_type: array<string, string>}|BlockRendererInterface $blockRendererOrCacheBlocks
-     */
     public function __construct(
-        BlockServiceManagerInterface $blockServiceManager,
```

Comment:
@VincentLanglet I noticed that we also need to remove the first argument :cry:  Otherwise its not used anymore at all...

So I guess we need another PR on 4.x to adjust the deprecation of the constructor signature? :confused:

Reply:
Indeed :'(

Reply:
will look into it :see_no_evil: